File: jja6312/2월12일/모음사전.java
class Solution {
    public int solution(String word) {
        // AEIOU
        char[] isAEIOU = { 'A', 'E', 'I', 'O', 'U' };
        int[] weight = { 781, 156, 31, 6, 1 };

        int answer = 0;
        for (int i = 0; i < word.length(); i++) {
            for (int j = 0; j < 5; j++) {
                if (word.charAt(i) == isAEIOU[j]) {
                    answer += 1 + j * weight[i];
                }
            }
        }

        return answer;
    }
}

File: zuzubibi/2월/2월 12일/모음사전.py
def solution(word):
    # 등비수열을 사용한 풀이법
    answer = 0
    for i,n in enumerate(word):
        answer += (5 **(5-i) -1) / (5-1) * "AEIOU".index(n) + 1
    return answer

#-----------------------------------------------------------------
from itertools import product
def anotherSolution(word):
    answer = []
    dic = ['A', 'E','I','O','U']

    for i in range(1,6):
        for per in product(dic, repeat=i):
            answer.append(''.join(per))
    answer.sort()

    return answer.index(word)+1

print(anotherSolution("AAAAE")) # 6
print(anotherSolution("AAAE")) # 10
print(anotherSolution("I")) #1563
print(anotherSolution("EIO")) #1189
